cli/cmd/generate_azure.go

```diff
@@ -0,0 +1,680 @@
+		"sub_ids",
+		[]string{},
+		`list of subscriptions to grant read access; format is id1,id2,id3`)
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.AdApplicationPassword,
+		"ad_pass",
+		"",
+		"password for an already existing active directory application")
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.AdApplicationId,
+		"ad_id",
+		"",
+		"id for an already existing active directory application")
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.AdServicePrincipalId,
+		"ad_pid",
+		"",
+		"principle id for an already existing active directory application")
+
+	generateAzureTfCommand.PersistentFlags().BoolVar(
+		&GenerateAzureCommandExtraState.TerraformApply, "terraform-apply", false, "run terraform apply for the generated hcl")
+
+}
+
+func promptAzureIntegrationNameQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Input{Message: QuestionAzureConfigName, Default: config.ConfigIntegrationName},
+			Checks:   []*bool{&config.Config},
+			Required: true,
+			Response: &config.ConfigIntegrationName,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionActivityLogName, Default: config.ActivityLogIntegrationName},
+			Checks:   []*bool{&config.ActivityLog},
+			Required: true,
+			Response: &config.ActivityLogIntegrationName,
+		},
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func promptAzureStorageAccountQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Confirm{Message: QuestionUseExistingStorageAccount, Default: config.ExistingStorageAccount},
+			Response: &config.ExistingStorageAccount,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionAwsRegion, Default: config.StorageLocation},
+			Checks:   []*bool{&config.ExistingStorageAccount},
+			Required: true,
+			Response: &config.StorageLocation,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionStorageAccountName, Default: config.StorageAccountName},
+			Checks:   []*bool{&config.ExistingStorageAccount},
+			Required: true,
+			Response: &config.StorageAccountName,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionStorageAccountResourceGroup, Default: config.StorageAccountResourceGroup},
+			Checks:   []*bool{&config.ExistingStorageAccount},
+			Required: true,
+			Response: &config.StorageAccountResourceGroup,
+		},
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func promptAzureSubscriptionQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Confirm{Message: QuestionEnableAllSubscriptions, Default: config.AllSubscriptions},
```

Comment:
Is there some debug left around somewhere for this? When I select no, I get this output. Unsure if this is intended...
```
▸ Enable all subscriptions? No
allSubscriptionsDisabled =>  true
```

Reply:
Yeah that was debug, that is removed !